File: python/queue/queue.py
class Queue:
    def __init__(self):
        self.items = []

    def is_empty(self):
        return len(self.items) == 0

    def front(self):
        return self.items[0]

    def back(self):
        return self.items[-1]

    def enqueue(self, item):
        self.items.append(item)

    def dequeue(self):
        return self.items.pop(0)

File: ruby/lib/queue/queue.rb
class Queue
  attr_accessor :items

  def initialize
    @items = []
  end

  def size
    items.size
  end

  def front
    items.first
  end

  def back
    items.last
  end

  def empty?
    items.empty?
  end

  def enqueue(item)
    items << item
  end

  def dequeue
    items.shift
  end
end